src/domains/issue/service/issue.service.impl.ts

```diff
@@ -1,55 +1,46 @@
-    if (project === null) {
-      throw new NotFoundException('Project');
+  /**
+   * Retrieves the total worked time in seconds for a given issue.
+   * This method calculates the total time tracked automatically and manually
+   * for the specified issue.
+   *
+   * @param {string} issueId - The ID of the issue to retrieve worked time for.
+   * @returns {Promise<number>} A promise that resolves to the total worked time in seconds.
+   * @throws {NotFoundException} If the specified issue cannot be found.
+   * @throws {ConflictException} If the calculated worked time is negative.
+   */
+  async getIssueWorkedSeconds(issueId: string): Promise<number> {
+    const issue: IssueDTO | null = await this.issueRepository.getById(issueId);
+    if (issue === null) {
+      throw new NotFoundException('Issue');
     }
+    let workedTime: number = 0;
 
-    const allIssuesData: IssueDataDTO[] = await this.adapter.adaptAllProjectIssuesData(project.providerId);
-    const filteredIssuesData: IssueDataDTO[] = await Promise.all(
-      allIssuesData.filter(async (issueData) => {
-        return (await this.issueRepository.getByProviderId(issueData.providerIssueId)) === null;
-      })
-    );
-    const integratedIssues: IssueDTO[] = await db.$transaction(async () => {
-      const finalIssues: IssueDTO[] = [];
-      for (const issueData of filteredIssuesData) {
-        const author: UserDTO | null = issueData.authorEmail !== null ? await this.userRepository.getByEmail(issueData.authorEmail) : null;
-        const assignee: UserDTO | null = issueData.assigneeEmail !== null ? await this.userRepository.getByEmail(issueData.assigneeEmail) : null;
-        const newIssue: IssueDTO = await this.issueRepository.create({
-          providerIssueId: issueData.providerIssueId,
-          authorId: author != null ? author.id : null,
-          assigneeId: assignee != null ? assignee.id : null,
-          projectId,
-          stageId: issueData.stage,
-          name: issueData.name,
-          title: issueData.title,
-          description: issueData.description,
-          priority: issueData.priority,
-          storyPoints: issueData.storyPoints,
-        });
-        finalIssues.push(newIssue);
-      }
+    const timeTrackings: TimeTrackingDTO[] = await this.eventRepository.getIssueTimeTrackingEvents(issueId);
+    workedTime = timeTrackings.reduce((result: number, timeTracking: TimeTrackingDTO) => {
+      return getTimeTrackedInSeconds(timeTracking.startTime, timeTracking.endTime) + result;
+    }, workedTime);
 
-      return finalIssues;
-    });
+    const manualTimeModifications: ManualTimeModificationDTO[] = await this.eventRepository.getIssueManualTimeModification(issueId);
+    workedTime = manualTimeModifications.reduce((result: number, time: ManualTimeModificationDTO) => result + time.timeAmount, workedTime);
 
-    return integratedIssues;
+    if (workedTime < 0) {
```

Comment:
This shouldn't be checked. If this happens, then it's because it was previous wrongly persisted in the database. So code shouldn't even get in this conditional. We have to make sure to save data correctly so this doesn't have to be checked ever.

Reply:
moved checking to the util method that calculates seconds (seconds cannot be negative)